plugins/inputs/logparser/logparser.go

```diff
@@ -128,11 +129,12 @@ func (l *LogParserPlugin) Start(acc telegraf.Accumulator) error {
 		DataFormat:             "grok",
 	}
 
-	var err error
-	l.GrokParser, err = parsers.NewParser(config)
+	parser := grok.Parser{}
+	err := parser.InitFromConfig(config)
```

Comment:
Can you please initialize the parser directly instead of using the old, to-be-removed, interface!?